pandas/core/frame.py

```diff
@@ -2259,9 +2262,14 @@ def to_markdown(
         result = tabulate.tabulate(self, **kwargs)
         if buf is None:
             return result
-        buf, _, _, _ = get_filepath_or_buffer(buf, mode=mode)
+        buf, _, _, should_close = get_filepath_or_buffer(
+            buf, mode=mode, storage_options=storage_options
+        )
         assert buf is not None  # Help mypy.
+        assert not isinstance(buf, str)
         buf.writelines(result)
+        if should_close:
+            buf.close()
```

Comment:
Should this be in a `try` / `finally` to ensure that the file is closed?

Reply:
It is done this way in other similar functions - except this one didn't implement close at all before.

Reply:
yeah we do it this way becase the opener isn't also close to where it needs to be closed (and some backends done't need closing)